flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/runtime/utils/AtomicRtasITCaseBase.java

```diff
@@ -0,0 +1,223 @@
+import org.apache.flink.types.Row;
+import org.apache.flink.util.FileUtils;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+/** The base case of atomic rtas ITCase. */
+public abstract class AtomicRtasITCaseBase extends TestLogger {
+
+    protected TableEnvironment tEnv;
+
+    protected abstract TableEnvironment getTableEnvironment();
+
+    @BeforeEach
+    void setup() {
+        tEnv = getTableEnvironment();
+        List<Row> sourceData = Collections.singletonList(Row.of(1, "ZM"));
+
+        TestCollectionTableFactory.reset();
+        TestCollectionTableFactory.initData(sourceData);
+
+        String sourceDDL = "create table t1(a int, b varchar) with ('connector' = 'COLLECTION')";
+        tEnv.executeSql(sourceDDL);
+        // clean data
+        TestSupportsStagingTableFactory.JOB_STATUS_CHANGE_PROCESS.clear();
+        TestSupportsStagingTableFactory.STAGING_PURPOSE_LIST.clear();
+    }
+
+    @Test
+    void testAtomicReplaceTableAs(@TempDir Path temporaryFolder) throws Exception {
+        commonTestForAtomicReplaceTableAs(
+                "atomic_replace_table", false, true, temporaryFolder.toFile());
+    }
+
+    @Test
+    void testAtomicReplaceTableAsWithDriverTableNotExists(@TempDir Path temporaryFolder)
+            throws Exception {
+        commonTestForAtomicReplaceTableAs(
+                "atomic_replace_table_not_exists", false, false, temporaryFolder.toFile());
+    }
+
+    @Test
+    void testAtomicCreateOrReplaceTableAs(@TempDir Path temporaryFolder) throws Exception {
+        commonTestForAtomicReplaceTableAs(
+                "atomic_create_or_replace_table", true, true, temporaryFolder.toFile());
+    }
+
+    @Test
+    void testAtomicCreateOrReplaceTableAsWithDriverTableNotExists(@TempDir Path temporaryFolder)
+            throws Exception {
+        commonTestForAtomicReplaceTableAs(
+                "atomic_create_or_replace_table_not_exists", true, false, temporaryFolder.toFile());
+    }
+
+    private void commonTestForAtomicReplaceTableAs(
+            String tableName,
+            boolean isCreateOrReplace,
+            boolean createDriverTable,
+            File tmpDataFolder)
+            throws Exception {
+        if (createDriverTable) {
+            tEnv.executeSql("create table " + tableName + " (a int) with ('connector' = 'PRINT')");
+        }
+
+        tEnv.getConfig().set(TableConfigOptions.TABLE_RTAS_CTAS_ATOMICITY_ENABLED, true);
+        String dataDir = tmpDataFolder.getAbsolutePath();
+        String sqlFragment =
```

Comment:
nit:
I think the code lines
` String sqlFragment =
                isCreateOrReplace
                        ? " create or replace table " + tableName
                        : " replace table " + tableName;`
can be extracted to a method.